akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala

```diff
@@ -1093,16 +1093,18 @@ trait FlowOps[+Out, +Mat] {
    *
    * See also [[FlowOps.splitAfter]].
    */
-  def splitWhen(p: Out ⇒ Boolean): SubFlow[Out, Mat, Repr, Closed] = {
+  def splitWhen(p: Out ⇒ Boolean, eagerClose: Boolean = false): SubFlow[Out, Mat, Repr, Closed] = {
```

Comment:
What do you thing about the signature? Positional elements after lambda look kinda ugly, so maybe an overload like:
```scala
def splitWhen(eagerClose: Boolean)(p: Out => Boolean)
```
should be introduced instead? This also has the advantage of being extendible with other options/flags in the future, while allowing to use a block syntax for lambda:
```scala
Source(0 to 10).splitWhen(eagerClose = true) { item =>
  ...
  ...
  ...
}
```